### PR TITLE
[4.1] Add label to the collapsible menu

### PR DESCRIPTION
**Pull request by brianteeman**
This PR takes the module title and uses it as an aria-label for the nav container of the default collapsible menu and the collapse-metismenu override in cassiopeia.

Pull Request for Issue #36839 

image

### CODE REVIEW

**Comment by chmst** [comment]
@Worti2 this looks good to me. Could you please test this with your user?

File: modules/mod_menu/tmpl/collapse-default.php [hunk 1]
```diff
@@ -15,7 +15,7 @@
 HTMLHelper::_('bootstrap.collapse');
 ?>
 
-<nav class="navbar navbar-expand-md">
+<nav class="navbar navbar-expand-md" aria-label="<?php echo $module->title;?>">
```
Comment (Quy) [inline modules/mod_menu/tmpl/collapse-default.php]:
```suggestion
<nav class="navbar navbar-expand-md" aria-label="<?php echo $module->title; ?>">
```

File: templates/cassiopeia/html/mod_menu/collapse-metismenu.php [hunk 2]
```diff
@@ -15,7 +15,7 @@
 HTMLHelper::_('bootstrap.collapse');
 ?>
 
-<nav class="navbar navbar-expand-lg">
+<nav class="navbar navbar-expand-lg" aria-label="<?php echo $module->title;?>">
```
Comment (Quy) [inline templates/cassiopeia/html/mod_menu/collapse-metismenu.php]:
```suggestion
<nav class="navbar navbar-expand-lg" aria-label="<?php echo $module->title; ?>">
```

**Comment by BertaOctech** [comment]
I have tested this item :white_check_mark: successfully on 0cf0845d2523d882b12bf8940e5639c3cbc14694<br /><br />I applied the patch and in fact the title of the module was included in the nav item as aria-label, as described. I marked the test as successful.

However, I wonder if this could be confusing. The title of the menu module is not always shown in the frontend. Either because the menu is in the menu position and not in the sidebar or because the author has decided to hide it. In this case you are using a label which is not visible in the screen. 

I guess we should choose appropriate names for our menu modules in any case.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/36900">issues.joomla.org/tracker/joomla-cms/36900</a>.</sub>

**Comment by RickR2H** [comment]
I have tested this item :white_check_mark: successfully on 0cf0845d2523d882b12bf8940e5639c3cbc14694<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/36900">issues.joomla.org/tracker/joomla-cms/36900</a>.</sub>

**Comment by RickR2H** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/36900">issues.joomla.org/tracker/joomla-cms/36900</a>.</sub>

**Comment by bembelimen** [comment]
Is this tested that the whole site does not break, if the title contains a quote ( " )?

**Comment by brianteeman** [comment]
> Is this tested that the whole site does not break, if the title contains a quote ( " )?

Good spot. I will update it to address that

**Comment by brianteeman** [comment]
Just to point out - nothing breaks with any extra " - its just the aria label that gets truncated - still needs fixing but not disruptive

**Comment by brianteeman** [comment]
Updated. 

If a module title contains a " then the aria label was truncated at the ". 

This update just strips the "

before aria-label="module title  " here"
after aria-label="module title here"

**Comment by chmst** [comment]
Back to pending<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/36900">issues.joomla.org/tracker/joomla-cms/36900</a>.</sub>

**Comment by brianteeman** [comment]
Please retest

File: modules/mod_menu/tmpl/collapse-default.php [hunk 3]
```diff
@@ -15,7 +15,7 @@
 HTMLHelper::_('bootstrap.collapse');
 ?>
 
-<nav class="navbar navbar-expand-md">
+<nav class="navbar navbar-expand-md" aria-label="<?php echo str_replace('"', '', $module->title); ?>">
```
Comment (Quy) [inline modules/mod_menu/tmpl/collapse-default.php]:
```suggestion
<nav class="navbar navbar-expand-md" aria-label="<?php echo htmlspecialchars($module->title, ENT_QUOTES, 'UTF-8'); ?>">
```

File: templates/cassiopeia/html/mod_menu/collapse-metismenu.php [hunk 4]
```diff
@@ -15,7 +15,7 @@
 HTMLHelper::_('bootstrap.collapse');
 ?>
 
-<nav class="navbar navbar-expand-lg">
+<nav class="navbar navbar-expand-md" aria-label="<?php echo str_replace('"', '', $module->title); ?>">
```
Comment (Quy) [inline templates/cassiopeia/html/mod_menu/collapse-metismenu.php]:
```suggestion
<nav class="navbar navbar-expand-md" aria-label="<?php echo htmlspecialchars($module->title, ENT_QUOTES, 'UTF-8'); ?>">
```

**Comment by Quy** [comment]
I have tested this item :white_check_mark: successfully on ce09280b469f45b2d0a9466dad71a9b75890e5fc<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/36900">issues.joomla.org/tracker/joomla-cms/36900</a>.</sub>

**Comment by pabloarias** [comment]
I have tested this item :white_check_mark: successfully on ce09280b469f45b2d0a9466dad71a9b75890e5fc<br /><br />Tested in Joomla 4.1.1-dev with PHP 8.0.15.

I agree with @BertaOctech: if you choose to not show module title, aria label is still showed. Maybe it shouldn't.

Thanks<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/36900">issues.joomla.org/tracker/joomla-cms/36900</a>.</sub>

**Comment by richard67** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/36900">issues.joomla.org/tracker/joomla-cms/36900</a>.</sub>

**Comment by laoneo** [comment]
Thanks!

**Pull request merged**